torch/ao/quantization/fx/prepare.py

```diff
@@ -774,10 +766,34 @@ def _recursive_maybe_replace_node_with_obs(
 
     new_args = []
     for old_arg in graph_output_node.args:
-        new_args.append(
-            _recursive_maybe_replace_node_with_obs(
-                old_arg, output_target_dtype, node_name_to_target_dtype,
-                qconfig_map, model, modules, graph))
+        if isinstance(old_arg, (list, tuple)):
+            for arg_idx, inner_node in enumerate(old_arg):
+                if arg_idx in output_quantized_idxs:
+                    output_target_dtype = output_quantized_idxs[arg_idx]
+                else:
+                    output_target_dtype = torch.quint8
+                new_args.append(
+                    _recursive_maybe_replace_node_with_obs(
+                        inner_node, output_target_dtype, node_name_to_target_dtype,
+                        qconfig_map, model, modules, graph))
+        elif isinstance(old_arg, dict):
+            results_dict = {}
+            for arg_idx, inner_v in enumerate(old_arg.items()):
```

Comment:
do we want to support {"key": dtype} or just treat the whole dictionary as one unit?